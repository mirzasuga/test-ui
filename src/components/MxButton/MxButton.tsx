import React from 'react';
import type { GestureResponderEvent } from 'react-native';
import { Text } from 'react-native';
import { TouchableOpacity } from 'react-native';

type MxButtonProps = {
  text: string;
  onPress: (event: GestureResponderEvent) => void;
};

const MxButton = function (props: MxButtonProps) {
  return (
    <TouchableOpacity onPress={props.onPress}>
      <Text>{props.text} </Text>
    </TouchableOpacity>
  );
};

MxButton.defaultProps = {
  text: 'render',
  onPress: (event: GestureResponderEvent) => {
    console.log({ event });
  },
};

export default MxButton;
