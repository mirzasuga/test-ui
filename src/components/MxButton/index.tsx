export { default as MxButton } from './MxButton';
