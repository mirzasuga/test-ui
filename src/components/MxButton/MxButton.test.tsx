import React from 'react';
import renderer from 'react-test-renderer';
import MxButton from './MxButton';

const onPressMock = jest.fn();

it('MxButton Render correctly', () => {
  const component = renderer.create(
    <MxButton text="Submit" onPress={onPressMock} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// it.todo('MxButton testing');
