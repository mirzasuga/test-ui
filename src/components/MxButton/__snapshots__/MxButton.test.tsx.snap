// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MxButton Render correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Submit
  </Text>
</View>
`;
